docs/source/data/errors.mdx

```diff
@@ -399,45 +393,110 @@ formatError: (formattedError, error) => {
+      // do something specific
+    }
+  },
+```
+
+However, if a _resolver_ throws the error, a `GraphQLError` is wrapped around that initially thrown error.
+
+The `unwrapResolverError` function either removes the `GraphQLError` wrapping from a resolver error or returns the error unaltered if it isn't from a resolver.
+
+So we can rewrite the above code snippet to work for errors thrown in and outside of resolvers, like so:
+
+```ts {7-9}
+import { unwrapResolverError } from '@apollo/server/errors';
+
+new ApolloServer({
+  formatError: (formattedError, error) => {
+    // unwrapResolverError removes the outer GraphQLError from
+    // errors thrown in resolvers
+    if (unwrapResolverError(error) instanceof CustomDBError) {
+      return { message: 'Internal server error' };
+    }
+  },
+});
+```
+
 > To make context-specific adjustments to the error received by `formatError` (such as localization or personalization), consider [creating a plugin](/apollo-server/integrations/plugins/) that uses the [`didEncounterErrors` lifecycle event](/apollo-server/integrations/plugins-event-reference/#didencountererrors) to attach additional properties to the error. These properties can be accessed from `formatError`.
 
 ### For Apollo Studio reporting
 
-You can use Apollo Studio to analyze your server's error rates. If you connect Apollo Server to Studio, all errors are sent to Studio by default. If you _don't_ want certain error information to be sent to Studio (either because the error is unimportant or because certain information is confidential), you can modify or redact errors entirely before they're transmitted.
+> **New in Apollo Server 4:** error details are [_not_ included in traces by default](../migration#usage-reporting-and-inline-trace-plugins-mask-errors-by-default). Instead, `<masked>` replaces each error's message, and the `maskedBy` error extension replaces all other extensions. The `maskedBy` extension includes the name of the plugin that performed the masking (`ApolloServerPluginUsageReporting` or `ApolloServerPluginInlineTrace`).
+
+You can use Apollo Studio to analyze your server's error rates. By default, the operations sent to Studio as detailed traces _don't_ contain error details.
 
-To accomplish this, you can provide a `rewriteError` function to the [usage reporting plugin](/apollo-server/api/plugin/usage-reporting/).
+If you _do_ want error information sent to Studio, you can choose to send to either send all errors unmodified or you can modify or redact specific errors before they're transmitted:
 
-> The usage reporting plugin is installed automatically with its default configuration if you provide an Apollo API key to Apollo Server. To define a custom `rewriteError` function, you need to install the plugin explicitly with a custom configuration, as shown in examples below.
+```
+new ApolloServer({
+  // etc.
+  plugins: [
+    ApolloServerPluginUsageReporting({
+      sendErrorsInTrace: { transform: rewriteError },
```

Comment:
It's `sendErrorsInTraces` (plural). (Many instances.)

You, uh, might want to fix the instances in migration.mdx while you're at it, oops.

Reply:
Eagle eyes! I'll correct those across the board 👍